runtime_lib/controller/main.cpp

```diff
@@ -820,6 +820,48 @@ void xaie_l2_dma_init(int col) {
   xaie::out32(xaie::getTileAddr(col, 0) + 0x00033008, 0xFF);
 }
 
+#ifdef ARM_CONTROLLER
+
+// Defining the NPI base and registers we use to reset the array
+uint64_t npi_base = 0xF70A0000UL;
+#define NPI_MASK_REG 0x0
+#define NPI_VAL_REG 0x4
+#define NPI_LOCK_REG 0xC
+void xaie_array_reset() {
+
+  // Getting a pointer to NPI
+  volatile uint32_t *npib = (volatile uint32_t *)(npi_base);
```

Comment:
You can avoid repeating the type in C++
```suggestion
  auto npib = (volatile uint32_t *)npi_base;
```

Reply:
Thanks for the pointer. Changed in [173d9da](https://github.com/Xilinx/mlir-air/pull/253/commits/173d9da6995d41a594bd9dc3e6d8972005afa95f).